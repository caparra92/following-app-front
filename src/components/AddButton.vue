<template>
    <button class="add-button">
        <ion-icon :icon="addOutline" size="large"></ion-icon>
    </button>
</template>
<script setup lang="ts">
import { IonIcon } from '@ionic/vue';
import { addOutline } from 'ionicons/icons';

</script>
<style scoped>
.add-button {
    position: fixed;
    bottom: 1.2em;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 2em;
    height: 2em;
    color: var(--ion-color-light);
    background: var(--ion-color-primary);
    border-radius: 50%;
    transition: all 0.40s ease-in-out;
}

.add-button:active {
    background: var(--ion-color-primary-shade);
    width: 2.2em;
    height: 2.2em;
}
</style>