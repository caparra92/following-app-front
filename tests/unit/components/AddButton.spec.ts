import { mount, VueWrapper } from "@vue/test-utils";
import AddButton from "@/components/AddButton.vue";

describe('AddButton tests', () => {

    let wrapper: VueWrapper;

    beforeEach(() => {
        wrapper = mount(AddButton);
    });

    test('Should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should verify that component exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });
});