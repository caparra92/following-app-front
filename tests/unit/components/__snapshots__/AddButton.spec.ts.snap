// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton tests Should match with snapshot 1`] = `
<button class="add-button">
  <ion-icon></ion-icon>
</button>
`;
